mesonbuild/compilers.py

```diff
@@ -2729,3 +2733,135 @@ def unix_compile_flags_to_native(self, args):
 
     def get_link_debugfile_args(self, targetfile):
         return []
+
+
+# Only tested on linux for ICC 17.0.0
+#    * object extraction does not compile (don't know) (25 object extraction fails)
```

Comment:
Yeah, I worked around it by changing that test and adding a new one which tests just this and is explicitly skipped for Intel.